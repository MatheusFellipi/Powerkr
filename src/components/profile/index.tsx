import { Avatar, Text, View } from "tamagui";

export default function Home() {
  return (
    <View>
      <Avatar
        circular
        size="$15"
      >
        <Avatar.Image
          accessibilityLabel="Cam"
          src="https://images.unsplash.com/photo-1548142813-c348350df52b?&w=150&h=150&dpr=2&q=80"
        />
        <Avatar.Fallback backgroundColor="$blue10" />
      </Avatar>
      <Text>Colaborador nome</Text>
      <Text>Departamento</Text>
    </View>
  );
}
