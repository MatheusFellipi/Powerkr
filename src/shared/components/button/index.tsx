import { Button, styled } from "tamagui";

export const BtnComponent = styled(Button, {
  fontSize: 14,
  borderRadius: 24,
  fontFamily: "InterBold",
  borderColor: "#4B5565",
  bg: "#202939",
  variants: {
    tabs: {
      activated: {},
      disabled: {
        borderColor: "$colorTransparent",
        bg: "$colorTransparent"
      }
    },
    styles: {
      login: {
        bg: "#F7B519",
        color: "#000"
      }
    }
  } as const
});
