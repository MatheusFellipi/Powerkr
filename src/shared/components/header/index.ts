import { Container } from "@shared/components/header/container";
import { Icon } from "@shared/components/header/icon";
import { TextComponent } from "@shared/components/textComponent";

export const HeaderComponent = {
  Text: TextComponent,
  Container: Container,
  Icon: Icon
};
