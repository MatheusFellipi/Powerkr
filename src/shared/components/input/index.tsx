import { useState } from "react";
import { Pressable, TextInputProps } from "react-native";
import { Icons } from "@assets/index";
import { Input, InputProps, Label, View } from "tamagui";

type Props = { label: string } & InputProps;

export const InputComponent = ({
  label,
  id,
  onChangeText,
  secureTextEntry,
  ...props
}: Props) => {
  const [showPassword, setShowPassword] = useState(secureTextEntry);
  return (
    <View>
      <Label
        color={"#CDD5DF"}
        htmlFor={id}
      >
        {label}
      </Label>
      <View
        fd="row"
        jc="space-between"
        ai="center"
        borderRadius={24}
        bg="#364152"
        borderWidth={1}
        borderColor="#4B5565"
      >
        <Input
          {...props}
          width={"90%"}
          borderRadius={24}
          secureTextEntry={showPassword}
          bg="$backgroundTransparent"
          borderColor="$backgroundTransparent"
        />
        {secureTextEntry && (
          <Pressable
            onPress={() => setShowPassword(!showPassword)}
            style={{
              width: "10%",
              paddingRight: 10
            }}
          >
            {showPassword ? (
              <Icons.EyeOn
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            ) : (
              <Icons.EyeOff
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            )}
          </Pressable>
        )}
      </View>
    </View>
  );
};
