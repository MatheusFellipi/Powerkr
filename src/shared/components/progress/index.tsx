import { CircularProgressComponent } from "@shared/components/progress/circular";

export const Progress = {
  Circular: CircularProgressComponent
};
