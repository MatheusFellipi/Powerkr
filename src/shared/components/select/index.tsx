import { useMemo, useState } from "react";
import { Icons } from "@assets/index";
import { Adapt, Select as S, Sheet } from "tamagui";

const items = [{ name: "Todos" }];

export function SelectComponent() {
  const [val, setVal] = useState("todos");
  return (
    <S
      id="food"
      value={val}
      onValueChange={setVal}
      disablePreventBodyScroll
    >
      <S.Trigger
        borderRadius={25}
        width={131}
        bg="$colorTransparent"
        borderColor="#4B5565"
        iconAfter={Icons.ArrowDown}
      >
        <S.Value placeholder="Something" />
      </S.Trigger>
      <Adapt
        when="sm"
        platform="touch"
      >
        <Sheet
          modal
          dismissOnSnapToBottom
          animationConfig={{
            type: "spring",
            damping: 20,
            mass: 1.2,
            stiffness: 250
          }}
        >
          <Sheet.Frame>
            <Sheet.ScrollView>
              <Adapt.Contents />
            </Sheet.ScrollView>
          </Sheet.Frame>
          <Sheet.Overlay
            animation="lazy"
            enterStyle={{ opacity: 0 }}
            exitStyle={{ opacity: 0 }}
          />
        </Sheet>
      </Adapt>
      <S.Content>
        <S.Viewport>
          <S.Group>
            {useMemo(
              () =>
                items.map((item, i) => {
                  return (
                    <S.Item
                      index={i}
                      key={item.name}
                      value={item.name.toLowerCase()}
                    >
                      <S.ItemText>{item.name}</S.ItemText>
                    </S.Item>
                  );
                }),

              [items]
            )}
          </S.Group>
        </S.Viewport>
      </S.Content>
    </S>
  );
}
