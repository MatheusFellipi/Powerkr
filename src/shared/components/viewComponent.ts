import { styled, View } from "tamagui";

export const ViewComponent = styled(View, {
  variants: {
    pin: {}
  } as const
});
