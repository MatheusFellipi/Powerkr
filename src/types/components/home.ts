type ApprovedProps = {
  id: number;
  title: string;
  date: Date;
  priority: string;
};

type MetricProps = {
  id: number;
  value: number;
  description: string;
};

type CardProfileProps = {
  name: string;
  company: string;
  urlImg: string;
};

type TabsComponentProps = {
  tab: string;
  handle: (value: string) => void;
};

type HomeComponentProps = {
  approved: ApprovedProps[];
  metric: MetricProps[];
  user: CardProfileProps;
};
