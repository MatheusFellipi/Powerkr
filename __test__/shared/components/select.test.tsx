import { SelectComponent } from "@shared/components/select";
import { render } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import config from "../../../tamagui.config";

describe("<App />", () => {
  it("renders correctly", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <SelectComponent />
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
