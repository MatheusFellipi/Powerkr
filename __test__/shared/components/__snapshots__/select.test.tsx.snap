// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "expanded": false,
      }
    }
    cursor="pointer"
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,0)",
        "borderBottomColor": "#4B5565",
        "borderBottomLeftRadius": 25,
        "borderBottomRightRadius": 25,
        "borderBottomWidth": 1,
        "borderLeftColor": "#4B5565",
        "borderLeftWidth": 1,
        "borderRightColor": "#4B5565",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#4B5565",
        "borderTopLeftRadius": 25,
        "borderTopRightRadius": 25,
        "borderTopWidth": 1,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "space-between",
        "maxWidth": "100%",
        "minHeight": 44,
        "overflow": "hidden",
        "paddingBottom": 10,
        "paddingLeft": 18,
        "paddingRight": 18,
        "paddingTop": 10,
        "width": 131,
      }
    }
  >
    <Text
      fontFamily="$body"
      lineBreakMode="clip"
      numberOfLines={1}
      pointerEvents="none"
      selectable={false}
      style={
        {
          "color": "#fff",
          "fontFamily": "Inter",
          "fontSize": 14,
          "letterSpacing": 0,
          "lineHeight": 23,
        }
      }
      suppressHighlighting={true}
    >
      Todos
    </Text>
    <View
      pointerEvents="none"
      style={
        {
          "height": 14,
          "minHeight": 14,
          "minWidth": 14,
          "width": 14,
        }
      }
    />
    <SvgMock
      size={14}
    />
  </View>,
  <View
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100001,
      }
    }
  >
    <View
      collapsable={false}
      disableClassName={true}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        {
          "height": "80%",
          "minHeight": "80%",
          "opacity": 0,
          "position": "absolute",
          "transform": [
            {
              "translateY": 10000.1,
            },
          ],
          "width": "100%",
          "zIndex": 100001,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#151515",
            "borderTopLeftRadius": 9,
            "borderTopRightRadius": 9,
            "flex": 1,
            "flexDirection": "column",
            "height": 0,
            "maxHeight": "100%",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          onMoveShouldSetResponder={[Function]}
          onResponderRelease={[Function]}
          onScroll={[Function]}
          onStartShouldSetResponder={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={8}
          style={
            {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              accessibilityLabelledBy=":r3:"
              accessibilityRole="none"
              style={
                {
                  "flexDirection": "column",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityLabelledBy=":r4:"
                accessibilityState={
                  {
                    "disabled": undefined,
                    "selected": true,
                  }
                }
                cursor="pointer"
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#151515",
                    "borderBottomColor": "#282828",
                    "borderLeftColor": "#282828",
                    "borderRightColor": "#282828",
                    "borderTopColor": "#282828",
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "justifyContent": "space-between",
                    "maxWidth": "100%",
                    "minHeight": 44,
                    "overflow": "hidden",
                    "paddingBottom": 10,
                    "paddingLeft": 18,
                    "paddingRight": 18,
                    "paddingTop": 10,
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <Text
                  fontFamily="$body"
                  id=":r4:"
                  lineBreakMode="clip"
                  numberOfLines={1}
                  style={
                    {
                      "color": "#fff",
                      "fontFamily": "Inter",
                      "fontSize": 14,
                      "letterSpacing": 0,
                      "lineHeight": 23,
                    }
                  }
                  suppressHighlighting={true}
                  userSelect="none"
                >
                  Todos
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          {
            "backgroundColor": "#151515",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "borderTopWidth": 0,
            "bottom": "-50%",
            "flex": 1,
            "flexDirection": "column",
            "height": 0,
            "left": 0,
            "maxHeight": "100%",
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "shadowOpacity": 0,
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </View>
  </View>,
]
`;
